manager/container.go

```diff
@@ -397,27 +399,44 @@ func (c *containerData) updateLoad(newLoad uint64) {
 	}
 }
 
-func (c *containerData) updateStats() error {
+func (c *containerData) updateStats() (nextCollectionTime time.Time, err error) {
+	nextCollectionTime, metrics, metricsErr := c.collectorManager.Collect()
+	if metricsErr != nil {
+		// Ignore errors if the container is dead.
+		if !c.handler.Exists() {
+			return nextCollectionTime, nil
+		}
+
+		// Metrics may be partially populated, push those before we return an error.
+		metricsErr = fmt.Errorf("%v, continuing to push metrics", metricsErr)
+	}
+	if metrics == nil {
```

Comment:
This condition should not have been here! Will remove it.

The intent was to  enable collection of metrics within 'housekeepingTick' so that the housekeeping time includes the time taken for metric collection.